src/lib/plugins/get-deps-from-plugin.ts

```diff
@@ -13,23 +13,36 @@ import {
 import analytics = require('../analytics');
 import { convertSingleResultToMultiCustom } from './convert-single-splugin-res-to-multi-custom';
 import { convertMultiResultToMultiCustom } from './convert-multi-plugin-res-to-multi-custom';
+import { processYarnWorkspaces } from './nodejs-plugin/yarn-workspaces-parser';
 
 const debug = debugModule('snyk-test');
 
+const processors = {
+  yarnWorkspaces: {
+    handler: processYarnWorkspaces,
+    files: ['package.json'],
+  },
+  allProjects: {
+    handler: getMultiPluginResult,
+    files: AUTO_DETECTABLE_FILES,
+  },
+};
+
 // Force getDepsFromPlugin to return scannedProjects for processing
 export async function getDepsFromPlugin(
   root: string,
   options: Options & (TestOptions | MonitorOptions),
 ): Promise<pluginApi.MultiProjectResult> {
   let inspectRes: pluginApi.InspectResult;
 
-  if (options.allProjects) {
+  if (options.yarnWorkspaces || options.allProjects) {
```

Comment:
I want to have something like the following to be clear that this is the path for multiple projects, but it doesn't look quite right to me
```suggestion
  if (options.keys().some(multiProjectProcessors.keys())
```